### DISCUSSION TITLE
What are the expected values for privacy_level in a sponsorship webhook?

### DISCUSSION THREAD

**Discussion by justdave** [community/community#134994] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Note to moderators: this is a question about a Webhook. Please leave it in the "API and Webhooks" category. Nobody in "Sponsors" will know the answer (you already tried that and it expired because nobody answered it). In fact, this probably needs a GitHub developer to answer it, as random people using the API probably aren't going to know.

The documentation shows a property called `privacy_level` in the sponsorship webhook, and defines it as a string, but does not define what the possible values are.  If I'm relaying sponsors events to somewhere public with my webhook, then I'll want to redact the sponsor information if that privacy_level is set for a private donor. Or rather, only display the sponsor info if they've made it public.

I found a (redacted) sample in this forum showing `"privacy_level": "private"` for a private donor, but haven't found any samples of a public one.  I would rather choose to show the info when it's public rather than hiding it when it's private, just in case some other category is added later that should still be private.

**Comment by edwinm** [comment, marked as answer]
There are only two possible values: PRIVATE and PUBLIC.

From the GraphQL documentation:

## The privacy of a sponsorship.

### Values for SponsorshipPrivacy

**PRIVATE**

Private.

**PUBLIC**

Public.

See [SponsorshipPrivacy](https://docs.github.com/en/graphql/reference/enums#sponsorshipprivacy)

**Reply by justdave** [reply]
Thanks. I had not run across the separate GraphQL docs, and that indeed fills in a lot of blanks that aren't in the main webhook docs.

**Comment by Chaitanya029** [comment]
Hi @justdave 

Thank you for the note to moderators, and I'm happy to help you with your question about the privacy_level property in the sponsorship webhook.

I understand your concern about redacting sponsor information based on the privacy_level setting. Unfortunately, the GitHub documentation doesn't provide explicit values for the privacy_level property.

However, based on your research, it seems that "private" is a possible value for private donors. To determine the value for public donors, I would recommend reaching out to GitHub support or a GitHub developer, as you mentioned, to get an authoritative answer.

In the meantime, you could consider implementing a whitelist approach, where you only display sponsor information if the privacy_level is explicitly set to a known public value (e.g., "public"). This way, you can ensure that you're not accidentally exposing private donor information.
